packages/datadog-plugin-langchain/src/handlers/chain.js

```diff
@@ -0,0 +1,48 @@
+'use strict'
+
+const LangChainHandler = require('./default')
+
+class LangChainChainHandler extends LangChainHandler {
+  getSpanStartTags (ctx) {
+    const tags = {}
+
+    if (!this.isPromptCompletionSampled()) return tags
+
+    let inputs = ctx.args?.[0]
+    inputs = Array.isArray(inputs) ? inputs : [inputs]
+
+    for (const idx in inputs) {
+      const input = inputs[idx]
+      if (typeof input !== 'object') {
+        tags[`langchain.request.inputs.${idx}`] = this.truncate(input)
+      } else {
+        for (const [key, value] of Object.entries(input)) {
+        // these are mappings to the python client names, ie lc_kwargs
+        // only present on BaseMessage types
+          if (key.includes('lc')) continue
+          tags[`langchain.request.inputs.${idx}.${key}`] = this.truncate(value)
+        }
+      }
+    }
+
+    return tags
+  }
+
+  getSpanEndTags (ctx) {
+    const tags = {}
+
+    if (!this.isPromptCompletionSampled()) return tags
+
+    let outputs = ctx.result
+    outputs = Array.isArray(outputs) ? outputs : [outputs]
```

Comment:
Are chain outputs returned as a list, or is this to be compatible with tracing chain.batch()?

Reply:
This is to be compatible with `batch`. I think in general chains could have `any` type, which is dynamic with respect to the chain output parser.